### PR TITLE
Improving visualHFSM

### PR DESCRIPTION
**Pull request by reysam93**
Hello, I've been a while working in visualHFSM and this is what I have already done.
I have added a runtime GUI that informs of the state of the automata, and I have added the possibility of generate python code (but this doesn't have the runtime GUI because it isn't finish).

I have also solved the bugs mentioned in the issue #183 , so it should be closed (I don't know how to do this)
I had created another pull request, but I have just seen that it has a bug so I have closed it and created this.

### CODE REVIEW

File: src/stable/components/visualHFSM/generate.cpp [hunk 1]
```diff
@@ -478,3 +658,424 @@ std::string Generate::getCppName () {
...
+		self.%1%Prx = %2%Prx.checkedCast(%1%)\n\
+		if(not self.%1%Prx):\n\
+			raise Exception('invalid proxy automata.%2%.Proxy')\n\
+		print '%1% connected'\n\
+\n");
+	for ( std::list<IceInterface>::iterator listInterfacesIterator = this->listInterfaces->begin();
+			listInterfacesIterator != this->listInterfaces->end(); listInterfacesIterator++ ) {
+		std::string iName = listInterfacesIterator->getName();
+		std::string iInterface = listInterfacesIterator->getInterface();
+		std::cout<<boost::str(fmt_iConnect % iName % iInterface);
```
Comment (reysam93) [inline src/stable/components/visualHFSM/generate.cpp]: This line should be:

this->fs << boost::str(fmt_iConnect % iName % iInterface);

Not to std::cout

**Comment by varhub** [comment]
Totally my fault.
I can do another pull request
-or-
You can fix it and push to master

This pull request will be updated (because relies on reysam93:master)
Reopen #210 will also gather last changes ;)

**Comment by reysam93** [comment]
Ok, I have fixed it and pushed it, so now should be fine.

**Comment by fqez** [comment]
Let me know when the component is ready so I can test it again and merge it! 

Thanks!

**Comment by reysam93** [comment]
The component should be ready now. Sorry but I fixed it and forgot completely about notify it.

**Comment by fqez** [comment]
It's ok. I will test it most likely tomorrow. 

Thank you very much!

**Comment by fqez** [comment]
Tested!

**Pull request merged**